### ISSUE TITLE
Subpasses example segfaults

### ISSUE THREAD

**Issue by illeatmyhat** [GPUOpen-LibrariesAndSDKs/V-EZ#70]
tl;dr, running the subpasses example will result in a segfault, because the first color attachment passed to the second subpass is `VK_ATTACHMENT_UNUSED`.
https://github.com/GPUOpen-LibrariesAndSDKs/V-EZ/blob/29d2229e44ee692f74a33b698c80273ebcc67c3e/Samples/Data/Shaders/Subpasses/Subpass0.frag#L13-L15
The first subpass will output to `attachments[1:3]`, but `attachments[0]` is unused.

https://github.com/GPUOpen-LibrariesAndSDKs/V-EZ/blob/29d2229e44ee692f74a33b698c80273ebcc67c3e/Samples/Data/Shaders/Subpasses/Subpass1.frag#L4-L8
This arrangement of layouts results in a call to `ConvertVkRenderPassCreateInfoToV2KHR` in the Vulkan SDK, executing the line:
```C++
const auto format = out_struct->pAttachments[input_attachment.attachment].format;
```
which causes a segfault because `input_attachment.attachment == VK_ATTACHMENT_UNUSED == ULONGINT_MAX`

Changing `Subpass0.frag` to
```glsl
layout(location = 0) out vec4 redChannel;
layout(location = 1) out vec4 greenChannel;
layout(location = 2) out vec4 blueChannel;
```
and `Subpass1.frag` to
```glsl
layout(input_attachment_index = 0, set = 0, binding = 0) uniform subpassInput redChannel;
layout(input_attachment_index = 1, set = 0, binding = 1) uniform subpassInput greenChannel;
layout(input_attachment_index = 2, set = 0, binding = 2) uniform subpassInput blueChannel;

layout(location = 3) out vec4 combinedChannels;
```
will fix the segfault, but result in a black screen. :boom:

**Comment by illeatmyhat** [comment]
Fixed by cleaning and rebuilding, checking out original code.
